packages/taco/src/conditions/context/context.ts

```diff
@@ -168,6 +173,38 @@ export class ConditionContext {
     return requestedParameters;
   }
 
+  public addCustomContextParameterValues(
+    customParameters: Record<string, CustomContextParam>,
+  ) {
+    Object.keys(customParameters).forEach((key) => {
+      this.validateCustomContextParameter(key);
+      this.customParameters[key] = customParameters[key];
+    });
+  }
+
+  public addAuthProvider(contextParam: string, authProvider: AuthProvider) {
+    if (!USER_ADDRESS_PARAMS.includes(contextParam)) {
+      throw new Error(
+        ERR_AUTH_PROVIDER_NOT_NEEDED_FOR_CONTEXT_PARAM(contextParam),
+      );
+    }
+
+    if (contextParam === USER_ADDRESS_PARAM_DEFAULT) {
+      if (authProvider instanceof EIP4361AuthProvider) {
+        this.authProviders[contextParam] = authProvider;
+        return;
+      }
+    } else if (contextParam === USER_ADDRESS_PARAM_EXTERNAL_EIP4361) {
+      if (authProvider instanceof SingleSignOnEIP4361AuthProvider) {
+        this.authProviders[contextParam] = authProvider;
+        return;
+      }
+    }
+    throw new Error(
+      ERR_INVALID_AUTH_PROVIDER_TYPE(contextParam, typeof authProvider),
+    );
```

Comment:
We can avoid nesting on line 192 if we throw a bit earlier, just like on line 186

Reply:
✔️